src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2104,22 +2104,16 @@ private void observeNew(@NonNull SCMHead head, @NonNull SCMRevision revision, @N
                 throw new IllegalStateException(
                     "Name of created project " + project + " did not match expected " + encodedName);
             }
-            // HACK ALERT
-            // ==========
-            // We don't want to trigger a save, so we will do some trickery to ensure that observer.created(project)
-            // performs the save
-            BulkChange bc = new BulkChange(project);
-            try {
-                project.setDisplayName(getProjectDisplayName(project, rawName));
-            } catch (IOException e) {
-                // ignore even if it does happen we didn't want a save
-            } finally {
-                bc.abort();
-            }
             // decorate contract is to ensure it does not trigger a save
             _factory.decorate(project);
             // ok it is now up to the observer to ensure it does the actual save.
-            observer.created(project);
+            try (BulkChange bc = new BulkChange(project);) {
+                observer.created(project);
+                project.setDisplayName(getProjectDisplayName(project, rawName));
+                bc.commit();
+            } catch (IOException e) {
+                // Ignored
+            }
```

Comment:
this change fixed the tests. the observer.created is what creates the `RunMap` that was null before. (noticed by walking back through a stacktrace of a successful one to see where it was created).

I'm not 100% sure on if there's any impact here and why it was so keen to avoid saving / why a bulk change couldn't be used around everything...